packages/studio-be/src/studio/cloud/types.ts

```diff
@@ -0,0 +1,68 @@
+type RuntimeStatus =
+  | 'ACTIVE'
+  | 'DELETED'
+  | 'BLOCKED'
+  | 'UPDATING'
+  | 'DELETING'
+  | 'BLOCKING'
+  | 'PENDING'
+  | 'STARTING'
+  | 'FAILED'
+
+export interface Introspect {
+  userId: string
+  runtimeName: string
+  runtimeStatus: RuntimeStatus
+  botName: string
+  botId: string
+}
+
+export type PersonalAccessToken = string
+export type OAuthAccessToken = string
+
+export interface Principals {
+  botId?: string
+  userId?: string
+  clientId?: string
+}
+
+const scopes = ['messaging', 'nlu'] as const
+type Scope = typeof scopes[number]
+
+interface ApiKey {
+  id: string
+  secret: string
+  scopes: readonly Scope[]
+}
+
+export interface Bot {
+  id: string
+  runtimeName: string
+  workspaceId: string
+  name: string
+  apiKey: ApiKey
+  configurations: BotConfiguration | null
+  filename: string | null
+  archive: Buffer | null
+  createdOn: string
+  updatedOn: string
+  importedOn: string | null
+}
+
+interface BotConfiguration {
+  cloud: BotCloudConfiguration
+  messaging: BotMessagingConfiguration
+}
+
+interface BotCloudConfiguration {
+  clientId: string
+  clientSecret: string
+}
+
+// TODO: create a type for input messaging config and output. input doesnt have webhook token
```

Comment:
What do you mean by input and output here?

Reply:
Don't know, I didn't write this comment